third-party/rsutils/src/byte-manipulation.cpp

```diff
@@ -0,0 +1,36 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include <rsutils/byte-manipulation.h>
+
+namespace rsutils
+{
+    namespace byte
+    {
+        // A helper function that takes a byte and reverses its bits
+        // e.g. intput: 00110101  output: 10101100
+
+        // First the left four bits are swapped with the right four bits.
+        // Then all adjacent pairs are swapped and then all adjacent single bits.
+        // This results in a reversed order.
+        unsigned char reverse_bits(unsigned char b)
+        {
```

Comment:
Please add a small non live unit test for this utility.
Like we do for other number utilities